tensorboard/webapp/core/views/page_title_test.ts

```diff
@@ -0,0 +1,158 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component, NO_ERRORS_SCHEMA} from '@angular/core';
+import {TestBed} from '@angular/core/testing';
+import {Store} from '@ngrx/store';
+import {provideMockStore, MockStore} from '@ngrx/store/testing';
+
+import {State} from '../state';
+import {getEnvironment} from '../store';
+import {RouteKind} from '../../app_routing/types';
+import {buildExperiment} from '../../experiments/store/testing';
+import {
+  getRouteKind,
+  getExperimentIdsFromRoute,
+  getExperiment,
+} from '../../selectors';
+import {TB_BRAND_NAME} from '../types';
+
+import {PageTitleModule} from './page_title_module';
+import {PageTitleComponent, TEST_ONLY} from './page_title_component';
+import {PageTitleContainer} from './page_title_container';
+
+describe('page title test', () => {
+  let store: MockStore<State>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [PageTitleComponent, PageTitleContainer],
+      providers: [provideMockStore()],
+      schemas: [NO_ERRORS_SCHEMA],
+    }).compileComponents();
+
+    store = TestBed.inject<Store<State>>(Store) as MockStore<State>;
+    store.overrideSelector(getRouteKind, RouteKind.EXPERIMENTS);
+    store.overrideSelector(getExperimentIdsFromRoute, []);
+    store.overrideSelector(getExperiment, null);
+    store.overrideSelector(getEnvironment, {
+      data_location: 'my-location',
+      window_title: '',
+    });
+  });
+
+  it('uses window_title as page title if given', () => {
+    const spy = spyOn(TEST_ONLY.utils, 'setDocumentTitle');
+    store.overrideSelector(getRouteKind, RouteKind.EXPERIMENT);
+    store.overrideSelector(getExperimentIdsFromRoute, ['123']);
+    store.overrideSelector(
+      getExperiment,
+      buildExperiment({
+        name: 'I will be overwritten by the window_title',
+      })
+    );
+    store.overrideSelector(getEnvironment, {
+      data_location: 'my-location',
+      window_title: 'I am the real title',
+    });
+    const fixture = TestBed.createComponent(PageTitleContainer);
+    fixture.detectChanges();
+
+    expect(spy).toHaveBeenCalledWith('I am the real title');
+  });
+
+  it('includes experiment name in tab title for experiment routes', () => {
+    const spy = spyOn(TEST_ONLY.utils, 'setDocumentTitle');
+    store.overrideSelector(getRouteKind, RouteKind.EXPERIMENT);
+    store.overrideSelector(getExperimentIdsFromRoute, ['123']);
+    store.overrideSelector(
+      getExperiment,
+      buildExperiment({
+        name: 'All you need is TensorBoard',
+      })
+    );
+    const fixture = TestBed.createComponent(PageTitleContainer);
+    fixture.detectChanges();
+
+    expect(spy).toHaveBeenCalledWith(
+      'All you need is TensorBoard - TensorBoard'
+    );
+  });
+
+  it('uses `Tensorboard` as default tab title', () => {
```

Comment:
Also, could we also add a test case covering the comparison route, specifically?  Right now it just shows the brand name, but might be worth considering something fancier later.

Reply:
SG, added a test case :)